### DISCUSSION TITLE
SFTP Reader Key Issue

### DISCUSSION THREAD

**Discussion by rjkroll** [nextgenhealthcare/connect#4698] [Q&A]
Hi,

I have a channel that is a SFTP reader.   Previously, we just used a username/password to login.

Now, the client provide two files:

- newkey.openssh
- newkey.secsh

This is my first time really using keys in Mirth and I am not sure what to do.   They did not provide any password or anything else, other then saying we had to update our processes.

In WinSCP, I was able to add the .openssh to the settings of this connection and I was able to login.   Can anyone provide my some guidance any what I would need to do to get the Mirth channel SFTP reader working again?

I have one other SFTP Reader where I added a .ppk file to the 'Wrench' Icon in that Connector and it worked, but I am not sure what to do with these two formats (.openssh/.secsh).  Do I need to convert one of them first before using in MIrth?

Thank you in advance for any help or tips

***********************************
2)	File Transfer Admin Actions
a)	Whitelisting - If you whitelist …
i)	The IP address will remain the same for bctransfer.bcbsal.org.
b)	SFTP - If you utilize SFTP …
i)	SSH Fingerprint Keys
(1)	Make sure you add this SSH Fingerprint key to your known_hosts file.
(2)	Add, do not replace the existing fingerprint key, as the existing key will be needed for accessing that legacy environment for 30 days after the migration.

ii)	SSH Authentication Keys
(1)	The attached SSH Key is for the SFTP Fingerprint security check that occurs on your system before the connection is allowed to continue to the authentication phase.
(2)	Any SSH authentication keys your entity may also have setup with Blue Cross Blue Shield of Alabama for your account logins will not be affected by this change.  Only the fingerprint keys will be affected.
c)	Ciphers
i)	The ciphers you utilize today are still supported in the new environment.
"

**Comment by jonbartels** [comment]
What version of MC are you using?

It sounds like your data source wants you to use key based auth. I know 3.9 (possibly older releases) supports key based authentication. It's in the user guide under "Advanced SFTP Options" on page 319. 

Forum threads are also a good resource for historical info - https://forums.mirthproject.io/search?q=sftp+ssh+key&searchJSON=%7B%22keywords%22%3A%22sftp+ssh+key%22%7D

**Comment by rjkroll** [comment]
We are on Mirth 3.7.0.    I dont have much experience with keys and security like this, but it looks like after I opened the session the first time  in WinSCP, and clicking 'Add Key' at the prompt, I went back into Mirth and clicked the 'Test Connection' button and now the interfaces are successful again.   I am assuming WinSCP added the key in such a way to the Windows Server that Mirth was able to access it (or maybe its the host file?).    I'm glad it's working, but I tried pointing to these keys in the Connector, converting them with Putty to .ppk (which I couldn't) and looking for the host file on the box, which I couldn't find.    Not sure what I would have done if I wasn't using WinSCP to add the key.

In the NextGen user's manual in the SFTP settings, it just says it accepts "a private/public keypair' but I had no clue what format's were acceptable or if I could just use what they sent me, point to the file, and be done with it or if I needed to convert using Putty or something first, to get it into a Mirth acceptable format.
